src/field_take_photo.c

```diff
@@ -0,0 +1,832 @@
+    TAKE_PHOTO_STATE_EXIT,
+} FieldTakePhotoTaskState;
+
+typedef enum FieldTakePhotoShutterState {
+    TAKE_PHOTO_SHUTTER_STATE_INIT,
+    TAKE_PHOTO_SHUTTER_STATE_DELAY_BEFORE,
+    TAKE_PHOTO_SHUTTER_STATE_FADE_OUT,
+    TAKE_PHOTO_SHUTTER_STATE_EXPOSURE,
+    TAKE_PHOTO_SHUTTER_STATE_FADE_IN,
+    TAKE_PHOTO_SHUTTER_STATE_WAIT_FADE_IN,
+    TAKE_PHOTO_SHUTTER_STATE_6,
+    TAKE_PHOTO_SHUTTER_STATE_DELAY_AFTER,
+} FieldTakePhotoShutterState;
+
+typedef struct FieldTakePhoto3 {
+    int state;
+    u8 filler_4[4];
+    FieldViewPhoto *parent;
+    u8 placeObjectCounter;
+} FieldTakePhoto3;
+ static BOOL FieldTask_ViewPhoto(TaskManager *taskManager);
+static int ViewPhotoFieldTask_Init(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_HandleAlbumSelection(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_LoadPhotoAndBeginRender(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_Cleanup(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_WaitInput(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_FadeInToPhoto(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_FadeOutFromPhoto(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static int ViewPhotoFieldTask_RestorePlayerOverworldPosition(FieldSystem *fieldSystem, TaskManager *taskManager, FieldViewPhoto *photo);
+static BOOL FieldTask_DoViewPhoto(TaskManager *taskManager);
+static LocalMapObject *createPartymonMapObject(MapObjectManager *objectMan, u16 species, u16 form, int gender, int direction, int x, int y, BOOL shiny);
+static LocalMapObject *createSpecialMapObject(MapObjectManager *objectMan, int spriteId, int direction, int x, int y, int localId);
+static void sub_0206AF78(FieldSystem *fieldSystem, PHOTO *photo);
+static void FieldTakePhoto_RemoveAllMapObjects(FieldTakePhoto *takePhoto);
+static void setCameraParam(Camera *camera);
+static void Photo_InitFromArcData(PHOTO *photo, FieldSystem *fieldSystem, u8 iconId, int mapId, int x, int y, int a6, int a7, u16 a8, u16 subjectObjId);
+static BOOL FieldTask_TakePhoto(TaskManager *taskManager);
+static void sub_0206B82C(PlayerAvatar *playerAvatar, u8 state, u8 gender);
+static void sub_0206B880(FieldSystem *fieldSystem, PHOTO *photo);
+static void drawCameraGfx(BgConfig *bgConfig, HeapID heapId);
+
+typedef struct Coord2U16 {
+    u16 x;
+    u16 y;
+} Coord2U16;
+
+static const Coord2U16 sSoloPhotoMonCoordOffsets = {  1, -1 };
+static const Coord2U16 sPhotoMonCoordOffsets[] = {
+    {  2,  0 },
+    {  1, -1 },
+    { -1, -1 },
+    {  3, -1 },
+    {  0, -2 },
+    {  2, -2 },
+};
+
+static PhotoCameraParam sCameraParam = {     .distance = FX32_CONST(666.922119140625),
```

Comment:
this should not be on the same line

Reply:
it'd look silly as a multiline

Reply:
disregard, vscode fucked up where the comment appears in the file